File: javascript/cells/src/cell-info.ts
export interface CellInfo {
  /**
   * User-friendly name for the device.
   */
  deviceName: string;
  /**
   * Width and height in density independent pixels.
   */
  width: number;
  height: number;
  /**
   * X/Y location relative to other cells.
   */
  x: number;
  y: number;
  /**
   * Server URL that the client uses.
   */
  server?: string;
}

export function cellCoordsValid(info: CellInfo) {
  return (
    !isNaN(info.x) &&
    !isNaN(info.y) &&
    !isNaN(info.width) &&
    !isNaN(info.height)
  );
}

File: android/app/src/main/java/com/tigeroakes/cellwall/client/device/CellInfo.kt
package com.tigeroakes.cellwall.client.device

import org.json.JSONObject

data class CellInfo(
  val deviceName: String,
  val density: Float,
  val widthPixels: Int,
  val heightPixels: Int
) {

  val width = (widthPixels / density).toInt()
  val height = (heightPixels / density).toInt()

  fun toJson() = JSONObject().apply {
    put("deviceName", deviceName)
    put("density", density)
    put("widthPixels", widthPixels)
    put("heightPixels", heightPixels)
  }
}
